File: 1.js
const fs = require("fs");

const numbers = fs
  .readFileSync("./1.txt")
  .toString()
  .split("\n")
  .map((x) => parseInt(x));

const sample = [199, 200, 208, 210, 200, 207, 240, 269, 260, 263];

const pt1 = ar => ar
  .map((_, i) => ar[i] < ar[i + 1])
  .filter(i=>i).length

const pt2 = ar => pt1(ar
  .map((_, i) => (ar[i] + 
                  ar[i + 1] + 
                  ar[i + 2]) || 0))

console.log(pt1(numbers));
console.log(pt2(numbers));

File: 1.clj
(ns aoc21-01
  (:require [clojure.string :as str]))

(def mem (map #(Integer/parseInt %) (str/split-lines (slurp "1.txt"))))

(defn pt1 [col] (count (filter true? (map #(apply < %) (partition 2 1 col)))))

(defn pt2 [col] (pt1 (map #(apply + %) (partition 3 1 col))))

(print "part 1:" (pt1 mem)
       "part 2:" (pt2 mem))

